src/taipy/config/global_app/global_app_config.py

```diff
@@ -111,7 +111,11 @@ def repository_properties(self, val):
 
     @property
     def properties(self):
-        return {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
+        return (
+            {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
+            if self._repository_properties
+            else self._DEFAULT_REPOSITORY_PROPERTIES
```

Comment:
Wait a minute.
 the fix has been implemented on `properties` getter instead of `repository_properties` getter. No?

Reply:
@jrobinAV 
ah no, we went with `repository_properties` for several reasons:
1. I remembered you saying that the self._ repository_properties should also be None, which I don't have a strong opinion on
2. Fixing this in repository_properties provides us with some benefit, namely no matter what functions or what ways the self._repository_properties is set to None, the repository_properties will always return an empty dict, this can help us escape from some potential problems

Reply:
oh dear, my bad :|

Reply:
no @toan-quach, I put it at `properties`, not `repository_properties` ...

Reply:
oh! 🤣

Reply:
🤣